__tests__/components/signup.test.js

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable no-undef */
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Signup } from '../../src/components/pages/Signup';
+
+const props = {
+  createAccount: jest.fn(),
+  ui: { loading: false },
+  auth: { signupSuccess: false },
+};
+
+const otherProps = {
+  ui: { loading: true },
+  createAccount: jest.fn(),
+  auth: { signupSuccess: false },
+};
+let signup;
+let signupLoading;
+describe('Signup Components tests...', () => {
+  beforeAll(() => {
+    signup = shallow(<Signup {...props} />);
+  });
+
+  it('Should render signup component', () => {
+    // console.log(signup.debug());
+    expect(signup).toMatchSnapshot();
+  });
+  it('Should find one form', () => {
+    const form = signup.find('form');
+    expect(form.length).toBe(1);
+  });
+  it('Should give initial state', () => {
+    // console.log(signup.state());
+    expect(signup.state()).toBeDefined();
+  });
+  describe('Input simulations tests...', () => {
+    it('Type in firstName field', () => {
+      const firstName = signup.find('input[name="firstName"]');
+      firstName.simulate('change', {
+        target: { value: 'Elie' },
+      });
+      expect(signup.state('firstName')).toEqual('Elie');
+    });
+    it('Type in lastName field', () => {
+      const lastName = signup.find('input[name="lastName"]');
+      lastName.simulate('change', {
+        target: { value: 'Mugenzi' },
+      });
+      expect(signup.state('lastName')).toEqual('Mugenzi');
+    });
+    it('Change the state on email typing', () => {
+      const email = signup.find('input[name="email"]');
+      email.simulate('change', {
+        target: { value: 'elie@me.com' },
+      });
+      expect(signup.state('email')).toEqual('elie@me.com');
+    });
+    it('Change the username state on type', () => {
+      const username = signup.find('input[name="username"]');
+      username.simulate('change', {
+        target: { value: 'elie' },
+      });
+      expect(signup.state('username')).toEqual('elie');
+    });
+    it('Should type the password and update its state', () => {
+      const password = signup.find('input[name="password"]');
+      password.simulate('change', {
+        target: { value: '123456' },
+      });
+      expect(signup.state('password')).toEqual('123456');
+    });
+    it('Should change confirmPassword state', () => {
+      const confirm = signup.find('input[name="confirm"]');
+      confirm.simulate('change', {
+        target: { value: '123456' },
+      });
+      expect(signup.state('confirmPassword')).toEqual('123456');
+    });
+    it('Should change DOB state', () => {
+      // console.log(signup.instance().props.createAccount);
```

Comment:
it would be better if you remove these console comments.

Reply:
Thanks @Diama1 , RESOLVED!